SunEngine.Core/Presenters/SearchPresenter.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using LinqToDB;
+using SunEngine.Core.DataBase;
+using SunEngine.Core.Services;
+
+namespace SunEngine.Core.Presenters
+{
+    public interface ISearchPresenter
+    {
+        Task<UserSearchInfoView[]> SearchByUserNameAndLink(string searchString);
+    }
+    
+    public class SearchPresenter : DbService, ISearchPresenter
+    {
+        public SearchPresenter(DataBaseConnection db) : base(db)
+        {
+        }
+
+        public Task<UserSearchInfoView[]>  SearchByUserNameAndLink(string searchString)
+        {
+            return db.Users.Where(x => searchString.Contains(x.UserName,StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Hm, I think that when searching for a user by name, it is logical to do x.UserName.Contains(searchString, StringComparison.OrdinalIgnoreCase).